spec/system/placements/support/schools/partner_providers/support_user_adds_a_partner_provider_spec.rb

```diff
@@ -1,245 +1,297 @@
   end
 
   private
 
-  def when_i_visit_the_partner_providers_page_for(school)
-    visit placements_school_partner_providers_path(school)
+  def given_providers_exist
+    @springfield_university = create(:placements_provider,
+                                     name: "Springfield University",
+                                     ukprn: "10101010",
+                                     urn: "101010",
+                                     email_addresses: ["reception@springfield.ac.uk"],
+                                     telephone: "0101 010 0101",
+                                     website: "http://www.springfield.ac.uk",
+                                     address1: "Undisclosed")
+    @springfield_university_user = create(:placements_user, providers: [@springfield_university])
+
+    @ogdenville_university = create(:provider,
+                                    name: "Ogdenville University",
+                                    ukprn: "00100010",
+                                    urn: "000010",
+                                    email_addresses: ["reception@ogdenville.ac.uk"],
+                                    telephone: "0000 000 1111",
+                                    website: "http://www.ogdenville.ac.uk",
+                                    address1: "1 Main Street")
+    @ogdenville_university_user = create(:placements_user, providers: [@ogdenville_university])
+
+    @shelbyville_university = create(:placements_provider, name: "Shelbyville University")
+
+    # Pre-existing partnership between school and Shelbyville University
+    @school = create(:school, :placements, name: "The London School")
+    create(:placements_partnership, school: @school, provider: @shelbyville_university)
+  end
 
-    then_i_see_support_navigation_with_organisation_selected
-    partner_providers_is_selected_in_secondary_nav
+  def and_the_school_partner_providers_flag_is_enabled
+    Flipper.add(:school_partner_providers)
+    Flipper.enable(:school_partner_providers)
   end
 
-  def when_i_click_on(text)
-    click_on text
+  def and_i_am_signed_in
+    sign_in_placements_support_user
   end
-  alias_method :and_i_click_on, :when_i_click_on
 
-  def and_i_enter_a_provider_named(provider_name)
-    fill_in "Add a provider", with: provider_name
+  def when_i_click_on_the_london_school
+    click_on "The London School"
   end
 
-  def then_i_see_a_dropdown_item_for(provider_name)
-    expect(page).to have_css(".autocomplete__option", text: provider_name, wait: 10)
+  def and_i_navigate_to_provider
+    within(primary_navigation) do
+      click_on "Providers"
+    end
+  end
+
+  def then_i_am_on_the_providers_index_page
+    expect(page).to have_current_path(placements_school_partner_providers_path(@school))
+    expect(page).to have_title("Providers you work with - Manage school placements - GOV.UK")
+    expect(primary_navigation).to have_current_item("Providers")
+    expect(page).to have_h1("Providers you work with")
```

Comment:
I tihnk there is a lot more stuff on this page that isnt referenced here - like the providers list or the empty providers list message